fleet/metrics.go

```diff
@@ -0,0 +1,212 @@
+package fleet
+
+import (
+	"context"
+	"github.com/go-kit/kit/metrics"
+	"time"
+)
+
+var _ AgentCommsService = (*commsMetricsMiddleware)(nil)
```

Comment:
So doing this as middleware is great for Core -> Edge metric, what we also need Edge -> Core metrics, which is done in handleMsgFromAgent .... so you need to find a way to instrument those as well (calls to handleCapabilities, handleHeartbeat, etc)